Sources/PointFree/Episode/Show.swift

```diff
@@ -45,29 +48,86 @@ let episodeResponse =
     }
 )
 
-let useCreditResponse =
-  fetchEpisodeForParam
+private func episodePageData(
+  currentUser: User?,
+  collectionSlug: Episode.Collection.Slug?,
+  episode: Episode,
+  permission: EpisodePermission,
+  subscriberState: SubscriberState
+) -> Either<EpisodePageData, NewEpisodePageData> {
+  if currentUser?.isAdmin == .some(true) {
+    let context: NewEpisodePageData.Context
+    if let collection = Episode.Collection.all.first(where: { $0.slug == collectionSlug }) {
+      context = .collection(collection)
+    } else {
+      context = .direct(
+        previousEpisode: Current.episodes().first(where: { $0.sequence == episode.sequence - 1 }),
+        nextEpisode: Current.episodes().first(where: { $0.sequence == episode.sequence + 1 })
+      )
+    }
+
+    return .right(NewEpisodePageData(
+      context: context,
+      date: Current.date,
+      episode: episode,
+      permission: permission,
+      subscriberState: subscriberState,
+      user: currentUser
+    ))
+  } else {
+    return .left(EpisodePageData(
+      permission: permission,
+      user: currentUser,
+      subscriberState: subscriberState,
+      episode: episode,
+      previousEpisodes: episode.previousEpisodes,
+      date: Current.date
+    ))
+  }
+}
+
+private func newOrLegacyEpisodeView(data: Either<EpisodePageData, NewEpisodePageData>) -> Node {
+  switch data {
+  case let .left(data): return episodeView(episodePageData: data)
+  case let .right(data): return newEpisodePageView(episodePageData: data)
+  }
+}
+
+func useCreditResponse<Z>(
```

Comment:
this is an upgrade of `useCreditResponse` so that it doesn't care about what data is tacked onto the end of the tuple.